### ISSUE TITLE
Why does baichuan-2 logits shift to larger values over training?

### ISSUE THREAD

**Issue by TissueC** [baichuan-inc/Baichuan2#174]
Hi, thanks for your valuable solution (z-loss) for logits shifting issue. I wonder have you ever explored the fundamental reasons under the phenomenon? 

This similar issue has been mentioned in DeepSpeed repo (refer to [issue 4379](https://github.com/microsoft/DeepSpeed/issues/4379) and [issue 4195](https://github.com/microsoft/DeepSpeed/issues/4195).
I think this may connect to the implementation of DeepSpeed.

Any commens would be highly appreciated. Thanks!
